src/library.js

```diff
@@ -649,7 +649,7 @@ addToLibrary({
 
     {{{ makeSetValue('daylight', '0', 'Number(winterOffset != summerOffset)', 'i32') }}};
 
-    var extractZone = (date) => date.toLocaleTimeString(undefined, {timeZoneName:'short'}).split(' ')[2];
```

Comment:
Would adding `'hour12': 'true'` to the options parameter be a more reliable way of getting this that doesn't depend on picking a particular locale? (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#options)

Reply:
I think picking a specific locale is the best way to do since that will fix the format.   Without that we don't really know what we are `split()`' ing since it depends completely on the locale, no?

Reply:
Hm, true. Mostly I was concerned that not every device is required to support every locale so there's no guarantee that the device supports en-US. I wonder if there's a way to use more options to constrain the format to be what we want.  

I guess another question, is the timezone name supposed to be localized (in which case we maybe want to let the local short form e.g. "PST" vs "GMT-8" leak through) nor not.

Reply:
> there's no guarantee that the device supports en-US

Oh.. i didn't realize it was ok for this locale to simply not exist.  Are you sure that is true?  If so, I agree we could use `'hour12': 'true'`

Reply:
`hour12` seems safer to me too, lgtm with that.

Reply:
well, https://github.com/emscripten-core/emscripten/pull/21585#issuecomment-2015439536 made me think of it, but I don't actually have any idea either. But in general, not all locales are required to be in all OS installs, are they? (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locales_argument 
also suggests maybe not?)